### DISCUSSION TITLE
Parsing and generating meson.build files

### DISCUSSION THREAD

**Discussion by bruchar1** [mesonbuild/meson#11444] [Q&A]
In my project, I want to write scripts to do the following:
- Check for errors in `meson.build` files (missing subdir, missing source files, etc.)
- Generate `meson.build` files (when creating a new library in the project, or when importing existing part of the project)
- Ensure consistency in the `meson.build` files format (indentations, blank lines, etc.)

meson provides `introspect --ast` and `rewrite` commands. However, those commands seem to only support partially what I want to achieve. What would be the best approach?

Should I try to improve meson to be able to:
- parse ast in arbitrary meson files
- convert ast to a meson.build file (that would be a meson file formatter)

Or would I be better to do this on my own, using regexp and string manipulation?

**Comment by kif** [comment]
Hi, I find this interesting since I would like to write unit-tests ensuring all files are effectively handled by meson ...
